server/planning/planning/planning_history_async_service.py

```diff
@@ -0,0 +1,203 @@
+            assigned_to = update.get("assigned_to")
+            user = update.get(
+                "proxy_user",
+                assigned_to.get("assignor_user", assigned_to.get("assignor_desk")),
+            )
+        history = {
+            "planning_id": item[ID_FIELD],
+            "user_id": user,
+            "operation": operation,
+            "update": update,
+        }
+
+        if operation == "create" and update.get("state", "") == "ingested":
+            history["operation"] = "ingested"
+
+        await self.create([history])
+
+    async def on_item_updated(
+        self, updates: dict[str, Any], original: PlanningResourceModel, operation: str | None = None
+    ):
+        item = deepcopy(original.to_dict())
+        if list(item.keys()) == ["_id"]:
+            diff = await self._remove_unwanted_fields(updates)
+        else:
+            diff = await self._changes(original, updates)
+            diff.pop("coverages", None)
+            if updates:
+                item.update(updates)
+
+        if len(diff.keys()) > 0:
+            operation = operation or "edited"
+            if original.get(LOCK_ACTION) in update_item_actions:
+                operation = original.get(LOCK_ACTION)
+                if original.get(LOCK_ACTION) == "assign_agenda":
+                    diff["agendas"] = [a for a in diff.get("agendas", []) if a not in original.get("agendas", [])]
+
+            if len(get_related_event_links_for_planning(diff, "primary")):
+                operation = "create_event"
+
+            await self._save_history(item, diff, operation)
+
+        await self._save_coverage_history(updates, original)
+
+    async def on_cancel(self, updates: dict[str, Any], original):
+        await self.on_item_updated(
+            updates,
+            original,
+            "planning_cancel" if original.get("lock_action") in ["planning_cancel", "edit"] else "events_cancel",
+        )
+
+    async def _get_coverage_diff(self, updates: dict[str, Any], original):
+        diff = {"coverage_id": original.get("coverage_id")}
+        cov_plan_diff = await self._changes(original.get("planning"), updates.get("planning"))
+
+        if cov_plan_diff:
+            diff["planning"] = cov_plan_diff
+
+        if original.get("news_coverage_status") != updates.get("news_coverage_status"):
+            diff["news_coverage_status"] = updates.get("news_coverage_status")
+
+        return diff
+
+    async def _save_coverage_history(self, updates: dict[str, Any], original):
+        """Save the coverage history for the planning item"""
+        item = deepcopy(original)
+        original_coverages = {c.get("coverage_id"): c for c in (original or {}).get("coverages") or []}
+        updates_coverages = {c.get("coverage_id"): c for c in (updates or {}).get("coverages") or []}
+        added, deleted, updated = [], [], []
+        planning_service = get_resource_service("planning")
```

Comment:
Looks like the old service is used here, for the [is_coverage_planning_modified](https://github.com/superdesk/superdesk-planning/blob/async/server/planning/planning/planning.py#L1251) and [is_coverage_assignment_modified](https://github.com/superdesk/superdesk-planning/blob/async/server/planning/planning/planning.py#L1265).

These two functions are quite small, and can probably be placed in a util file somewhere and used here (so we are not using the old legacy service)